File: clients/node/sdk/models/tag.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tag = void 0;
const createHistoryItemFromDiscriminatorValue_1 = require("./createHistoryItemFromDiscriminatorValue");
class Tag {
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    get additionalData() {
        return this._additionalData;
    }
    ;
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    set additionalData(value) {
        this._additionalData = value;
    }
    ;
    /**
     * Instantiates a new Tag and sets the default values.
     */
    constructor() {
        this._additionalData = {};
    }
    ;
    /**
     * Gets the following property value. The following property
     * @returns a boolean
     */
    get following() {
        return this._following;
    }
    ;
    /**
     * Sets the following property value. The following property
     * @param value Value to set for the following property.
     */
    set following(value) {
        this._following = value;
    }
    ;
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    getFieldDeserializers() {
        return {
            "following": n => { this.following = n.getBooleanValue(); },
            "history": n => { this.history = n.getCollectionOfObjectValues(createHistoryItemFromDiscriminatorValue_1.createHistoryItemFromDiscriminatorValue); },
            "name": n => { this.name = n.getStringValue(); },
            "url": n => { this.url = n.getStringValue(); },
        };
    }
    ;
    /**
     * Gets the history property value. The history property
     * @returns a HistoryItem
     */
    get history() {
        return this._history;
    }
    ;
    /**
     * Sets the history property value. The history property
     * @param value Value to set for the history property.
     */
    set history(value) {
        this._history = value;
    }
    ;
    /**
     * Gets the name property value. The name property
     * @returns a string
     */
    get name() {
        return this._name;
    }
    ;
    /**
     * Sets the name property value. The name property
     * @param value Value to set for the name property.
     */
    set name(value) {
        this._name = value;
    }
    ;
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    serialize(writer) {
        if (!writer)
            throw new Error("writer cannot be undefined");
        writer.writeBooleanValue("following", this.following);
        writer.writeCollectionOfObjectValues("history", this.history);
        writer.writeStringValue("name", this.name);
        writer.writeStringValue("url", this.url);
        writer.writeAdditionalData(this.additionalData);
    }
    ;
    /**
     * Gets the url property value. The url property
     * @returns a string
     */
    get url() {
        return this._url;
    }
    ;
    /**
     * Sets the url property value. The url property
     * @param value Value to set for the url property.
     */
    set url(value) {
        this._url = value;
    }
    ;
}
exports.Tag = Tag;

File: clients/csharp/sdk/Models/Tag.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System;
namespace MastodonClientLib.Models {
    public class Tag : IParsable {
        /// <summary>The following property</summary>
        public bool? Following { get; set; }
        /// <summary>The history property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public List<HistoryItem>? History { get; set; }
#nullable restore
#else
        public List<HistoryItem> History { get; set; }
#endif
        /// <summary>The name property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Name { get; set; }
#nullable restore
#else
        public string Name { get; set; }
#endif
        /// <summary>The url property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Url { get; set; }
#nullable restore
#else
        public string Url { get; set; }
#endif
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static Tag CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new Tag();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"following", n => { Following = n.GetBoolValue(); } },
                {"history", n => { History = n.GetCollectionOfObjectValues<HistoryItem>(HistoryItem.CreateFromDiscriminatorValue)?.ToList(); } },
                {"name", n => { Name = n.GetStringValue(); } },
                {"url", n => { Url = n.GetStringValue(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteBoolValue("following", Following);
            writer.WriteCollectionOfObjectValues<HistoryItem>("history", History);
            writer.WriteStringValue("name", Name);
            writer.WriteStringValue("url", Url);
        }
    }
}
